launchable/utils/launchable_client.py

```diff
@@ -102,15 +102,29 @@ def print_exception_and_recover(self, e: Exception, warning: Optional[str] = Non
             click.echo(click.style(warning, fg=warning_color), err=True)
 
     def is_fail_fast_mode(self) -> bool:
-        if 'fail_fast_mode' in self._workspace_state_cache:
-            return self._workspace_state_cache['fail_fast_mode']
-        # TODO: call api and set the result to cache
+        state = self._get_workspace_state()
+        return state.get('fail_fast_mode', False)
+
+    def is_enabled_pts_v2(self) -> bool:
+        state = self._get_workspace_state()
+        return state.get('state', "") == "HANDS_ON_LAB_V2"
+
+    def _get_workspace_state(self) -> dict:
+        """
+        Get the current state of the workspace.
+        """
+        if self._workspace_state_cache is not None:
+            return self._workspace_state_cache
         try:
             res = self.request("get", "state")
             if res.status_code == 200:
-                self._workspace_state_cache['fail_fast_mode'] = res.json().get('isFailFastMode', False)
-                return self._workspace_state_cache['fail_fast_mode']
+                state = res.json()
+                self._workspace_state_cache = {
+                    'state': state.get('state', ""),
+                    'fail_fast_mode': state.get('isFailFastMode', False)
```

Comment:
Is it enough to check only the existence for the value of 'isFailFastMode'? Is there a possibility that the value of 'isFailFastMode' could be false?

Reply:
Yes, the `isFailFastMode` field is bool type